include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -148,89 +148,88 @@ struct size_type<it_t>
 #endif // SEQAN3_DEPRECATED_310
 
 // see specialisation for ranges in core/type_traits/range.hpp
+//!\}
 
-// ----------------------------------------------------------------------------
-// iterator_tag
-// ----------------------------------------------------------------------------
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+#if SEQAN3_WORKAROUND_GCC_96070
+//!\cond
+template <typename it_t>
+struct iterator_category_tag
+{
+    using iterator_category = void;
```

Comment:
so if iterator_category is not defined we set it to void? That means this becomes not a valid iterator anymore with the concepts. So I always have to specify the iterator category for every iterator, or must make sure, that the iterator_traits is overloaded for the iterator and specifies the iterator_category. If I read the standard correctly only `ITER_CONCEPT` must derive from the respective iterator tag it models. So my read on this is that a c++20 iterator is well defined, if it defines an iterator_concept but not iterator category. Ah ok. So we always have to specify the iterator_concept for all our iterators and to make them compliant with legacy iterators we need to define iterator_category, unless it is not defined in the underlying iterator. Then the underlying iterator was already not a valid legacy iterator and we don't need to make it a valid one. Sorry for the brain dump. I just needed to understand this. Maybe you can verify or correct me if I am seeing it wrong.

Reply:
See https://github.com/seqan/seqan3/issues/1963#issuecomment-653820580 for an explanation.